### PR TITLE
fix: prevent duplicated destination prefixes for template filename paths

### LINKED ISSUE #1116: [BUG] MACROS: Created Notes from Template with wrong destination paths
Hi @chhoumann,

since a few weeks I have a bug with Quick-Add Macros:

**Describe the bug**
I created diffent Macros with User-Scripts to create notes based on a template.

I defined File name format:
03_Aufgabenmanagement/ToDos/{{VALUE:Extracted_Task_Path}}/{{VALUE:Extracted_Task_ID}}/Note - {{DATE:}} - {{VALUE:Random_ID}}.excalidraw

Settings Template:
Image

Settings Macro:

Image

{{VALUE:Extracted_Task_Path}}, {{VALUE:Extracted_ID_Task}} and Rondom ID = Value from UserScript and variables set correct.

**Expected behavior**

New note should be in folder:
03_Aufgabenmanagement/ToDos/W-Work/this_is_a_task/Note - 2026-02-23 - a39.excalidraw

but path is:
03_Aufgabenmanagement/ToDos/W-Tanso/03_Aufgabenmanagement/ToDos/W-Tanso/ein_neuer_taesk/Note - 2026-02-23 - w96.excalidraw

This directory of the beginning 03_Aufgabenmanagement/ToDos/W-Tanso/ is complete random.
Sometimes it is put und 00_Journaling/Daily_Notes/03_Aufgabenmanagement/ToDos/W-Tanso/ein_neuer_taesk/Note - 2026-02-23 - w96.excalidraw

QuickAdd - Random ID generiert: w96
VM7096:42 ======= QUICKADD YAML DEBUG =======
VM7096:43 Datei: 03_Aufgabenmanagement/ToDos/W-Tanso/Ein neuer Täsk.md
VM7096:45 
(index)
Wert im YAML
Key
Variable: {{VALUE:Extracted_Task_Path}}	'W-Tanso'	'Task_Path'
Variable: {{VALUE:Extracted_Task_ID}}	'ein_neuer_taesk'	'Task_ID'
Object
VM7096:50 ===================================
plugin:quickadd:74 QuickAdd: (LOG) TemplateEngine.createFileWithTemplate: Collected 0 template property variables for **03_Aufgabenmanagement/ToDos/W-Tanso/**03_Aufgabenmanagement/ToDos/W-Tanso/ein_neuer_taesk/Note - 2026-02-23 - w96.excalidraw.md

Log shows, that Scripts are working correct.

**Obsidian Debug Info (required for bug reports)**
SYSTEM INFO:
	Obsidian version: v1.11.7
	Installer version: v1.11.5
	Operating system: Windows 11 Enterprise 10.0.26200
	Login status: logged in
	Language: en
	Catalyst license: none
	Insider build toggle: off
	Live preview: on
	Base theme: adapt to system
	Community theme: Minimal v8.1.4
	Snippets enabled: 8
	Restricted mode: off
	Plugins installed: 32
	Plugins enabled: 32
		5: QuickAdd v2.11.0

### PR DESCRIPTION
**Pull request by chhoumann**
## Summary
Fix template choice destination path resolution when `Create in folder` is disabled and `File name format` already includes folders.

QuickAdd was always prepending Obsidian's resolved default note folder (`getNewFileParent`) to the formatted filename. If the formatted name was already a vault-relative path, this produced duplicated/incorrect prefixes.

This change:
- strips exact duplicate folder prefixes from formatted names
- treats slash-separated formatted names as explicit vault-relative paths when their first segment exists at vault root
- preserves existing default-folder behavior for plain filenames and relative subpaths

Closes #1116.

### Before fix (confirmed)
- Active note in `03_Aufgabenmanagement/ToDos/W-Tanso/...`
- File name format: `03_Aufgabenmanagement/ToDos/W-Work/this_is_a_task/Issue1116-before-{{DATE:YYYY-MM-DD}}.excalidraw`
- Created path:
  - `03_Aufgabenmanagement/ToDos/W-Tanso/03_Aufgabenmanagement/ToDos/W-Work/this_is_a_task/Issue1116-before-2026-02-23.excalidraw.md`
- Console log showed same duplicated destination in `TemplateEngine.createFileWithTemplate`.

### After fix (confirmed)
Same setup now creates:
- `03_Aufgabenmanagement/ToDos/W-Work/this_is_a_task/Issue1116-after-current-2026-02-23.excalidraw.md`

With Obsidian default note location set to fixed folder (`00_Journaling/Daily_Notes`), same format now creates:
- `03_Aufgabenmanagement/ToDos/W-Work/this_is_a_task/Issue1116-after-daily-2026-02-23.excalidraw.md`

No runtime errors were captured via `obsidian vault=dev dev:errors`.

## Tests
Added regression coverage in `src/engine/TemplateChoiceEngine.notice.test.ts`:
- full-path filename format is treated correctly (no duplicate prefix)
- plain filename still follows Obsidian default new-note location
- relative subpaths remain under default location

Executed:
- `bun run test src/engine/TemplateChoiceEngine.notice.test.ts`
- `bun run test`
- `bun run build-with-lint`

---

<a href="https://app.devin.ai/review/chhoumann/quickadd/pull/1118" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive tests for template destination path resolution covering vault-relative paths, leading-slash and deep subpaths, plain filenames, default-folder interactions, create-in-folder behavior, and root-edge cases.

* **Bug Fixes**
  * Improved template file path resolution so vault-relative names, duplicate-folder prefixes, relative subpaths, and create-in-folder scenarios produce correct destination paths.

### CODE REVIEW

File: src/engine/TemplateChoiceEngine.ts [hunk 1]
```diff
@@ -63,26 +63,33 @@ export class TemplateChoiceEngine extends TemplateEngine {
...
+					this.choice.name,
+				),
+				folderPath,
 			);
-			
+			const treatAsVaultRelativePath =
+				await this.shouldTreatFormattedNameAsVaultRelativePath(
+					formattedName,
+					folderPath,
+				);
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/engine/TemplateChoiceEngine.ts:91]: Addressed in commit 3297d54. stripDuplicateFolderPrefix now returns both the normalized name and a strippedPrefix flag, and shouldTreatFormattedNameAsVaultRelativePath immediately returns false when strippedPrefix is true. This prevents the default-folder prefix from being dropped after duplicate-prefix stripping. Added regression test: does not drop default-folder prefix after duplicate-prefix stripping, and revalidated in-app with projects/docs paths.

File: src/engine/TemplateChoiceEngine.ts [hunk 2]
```diff
@@ -304,6 +312,44 @@ export class TemplateChoiceEngine extends TemplateEngine {
...
+
+		const normalizedFileName = formattedName.trim();
+		if (!normalizedFileName.includes("/")) return false;
+		if (normalizedFileName.startsWith("./")) return false;
+		if (normalizedFileName.startsWith("/")) return true;
+
+		const slashCount = normalizedFileName.split("/").length - 1;
+		// Keep one-level subpaths (e.g. "tasks/note") relative to Obsidian's
+		// default folder, while treating deeper paths as vault-relative.
+		return slashCount >= 2;
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/engine/TemplateChoiceEngine.ts]: Valid point. Addressed in commit 81216de. I removed the slash-count heuristic and now classify vault-relative paths by checking the first segment against vault root entries via getAbstractFileByPath, requiring a TFolder. This keeps deep relative subpaths under the default folder when the root segment is missing and avoids treating root-level files as folder roots. Added regression tests for both cases: deep relative subpaths stay relative, and root-level files are not misclassified.
